pkg/cloud-provider/cloudapi/azure/azure_security.go

```diff
@@ -655,6 +703,40 @@ func (computeCfg *computeServiceConfig) getATGroupView(nepheControllerATSGNameTo
 	return enforcedSecurityCloudView, nil
 }
 
+// isValidAgSg finds if an ASG is in managed VNET, and it is a valid nephe created AddressGroup SG.
+func (computeCfg *computeServiceConfig) isValidAgSg(asgIDLowercase string,
+	asgIDToVnetID map[string]string) (string, string, error) {
+	// proceed only if ASG is associated with network interfaces of managed VNET.
+	vnetID, found := asgIDToVnetID[asgIDLowercase]
+	if !found {
+		return "", "", fmt.Errorf("asg not associated with managed network interfaces [%v]", asgIDLowercase)
+	}
+
+	// proceed only if ASG is a nephe created AddressGroup SG.
+	_, asgRgName, cloudAsgName, err := extractFieldsFromAzureResourceID(asgIDLowercase)
+	if err != nil {
+		return "", "", err
+	}
+
+	asgName, isAG, _ := securitygroup.IsNepheControllerCreatedSG(cloudAsgName)
+	if !isAG {
+		return "", "", fmt.Errorf("asg is not a nephe created AddressGroup SG [%v]", cloudAsgName)
```

Comment:
why you need to throw error if asg is not nephe created.. this function should just check if asg is nephe created or not

Reply:
This error is only for indication purpose on the caller side to figure out if CloudResource(used for creating sync content at later stage) should be created for the ASG or not . Not really flagging it anywhere.

Reply:
Added bool for the return value as discussed.